R/simulate.R

```diff
@@ -477,3 +477,107 @@ simulate_chain_stats <- function(n_chains,
 
   return(out)
 }
+
+#' Calculate transmission chain statistics across a range of scenarios and
+#' divide into intervals
+#'
+#' @details
+#' The `offspring_dist` argument is not equivalent to the `offspring_dist`
+#' argument in [simulate_chains()] and [simulate_chain_stats()], in those
+#' functions `offspring_dist` should be a `<function>`, whereas in
+#' `simulate_scenarios()` `offspring_dist` should be a vector of `character`
+#' strings which match function names.
+#'
+#' @inheritParams simulate_chains
+#' @param offspring_dist Offspring distribution: a `character` string  with the
+#' name of the random number generator function. For example those
+#' provided by R to generate random numbers from given distributions (e.g.,
+#' [rpois()] for Poisson). If the `character` string does not match
+#' a function then the function will error (see [match.fun()] for details).
+#' Examples that can be provided here are `"rpois"` for Poisson distributed
+#' offspring, `"rnbinom"` for negative binomial offspring.
+#' @param R_seq A `numeric` vector of reproduction number values to simulate
+#' the branching process for.
+#' @param k_seq A `numeric` vector of dispersion values to simulate the
+#' branching process for. Only applicable for `offspring_dist = "rnbinom"`.
+#' @inheritParams base::cut
+#' @param ... [dots] Extra arguments to be passed to [simulate_chain_stats()].
+#' If argument does not match one from [simulate_chain_stats()] the function
+#' will error.
+#'
+#' @return A `<data.frame>`.
+#' @export
+#'
+#' @examples
+#' simulate_scenarios(
+#'   statistic = c("size", "length"),
+#'   offspring_dist = c("rpois", "rnbinom"),
+#'   R_seq = seq(0.1, 0.5, 0.1),
+#'   k_seq = seq(0.1, 0.5, 0.1),
+#'   breaks = c(0, 2, 5, 10, 20, 50, 100, Inf)
+#' )
+#'
+#' # when R > 1 pass a stopping criteria to simulate_chain_stats()
+#' simulate_scenarios(
+#'   statistic = c("size", "length"),
+#'   offspring_dist = c("rpois", "rnbinom"),
+#'   R_seq = seq(0.5, 2.0, 0.5),
+#'   k_seq = seq(0.1, 0.5, 0.1),
+#'   breaks = c(0, 2, 5, 10, 20, 50, 100, Inf),
+#'   stat_threshold = 10
+#' )
+simulate_scenarios <- function(statistic,
+                               offspring_dist,
+                               R_seq,
+                               k_seq,
+                               breaks,
+                               ...) {
```

Comment:
Currently, this function simulates only one run for each scenario, so we will need to allow for multiple runs here. This is related to #41. I'm torn between having it here versus in the main simulation functions because users may not want to explore scenarios using this function but may want to run the main functions multiple times for one parameter set.

Reply:
Good point. I'll follow your lead on where you think this is best to implement.